Sources/ToucanSource/Bundle/SourceBundleRenderer.swift

```diff
@@ -143,15 +166,205 @@ public struct SourceBundleRenderer {
                 using: pipeline
             )
 
-            let assetPipelineResolver = AssetPipelineResolver(
+            func filterFilePaths(
+                from paths: [String],
+                input: Pipeline.Assets.Location
+            ) -> [String] {
+                paths.filter { filePath in
+                    guard let url = URL(string: filePath) else {
+                        return false
+                    }
+
+                    let path = url.deletingLastPathComponent().path
+                    let name = url.deletingPathExtension().lastPathComponent
+                    let ext = url.pathExtension
+
+                    let inputPath = input.path ?? ""
+                    let pathMatches =
+                        inputPath == "*" || inputPath.isEmpty
+                        || path == inputPath
+                    let nameMatches =
+                        input.name == "*" || input.name.isEmpty
+                        || name == input.name
+                    let extMatches =
+                        input.ext == "*" || input.ext.isEmpty
+                        || ext == input.ext
+                    return pathMatches && nameMatches && extMatches
+                }
+            }
+
+            let assetsPath = sourceBundle.config.contents.assets.path
+
+            for content in contents {
+                var assetsReady: Set<String> = .init()
+
+                var behaviors = pipeline.assets.behaviors
+                if behaviors.filter({ $0.id == "copy" }).isEmpty {
```

Comment:
Is it a good idea to add this behaviour by default? What about non-html pipelines? We just removed similar default behaviours for `css` and `js` files.

Reply:
Good point, I'll also make input & output optionals, so users can simply use:

```yaml
assets:
  behaviors:
    - id: copy
```